### PR TITLE
NSX-T integration

### PR DESCRIPTION
**Pull request by richardtief**
Implementation of the collection of NSX-T resources in the exporter. First of all, in this pull request only all NSX-T adapters and their associated management clusters. The metrics of the management clusters are the first ones that are needed. The possibility of extension is given.

### CODE REVIEW

File: resources/Resourceskinds.py [hunk 1]
```diff
@@ -80,3 +80,30 @@ def __init__(self, name, uuid):
...
+    def __init__(self, name, uuid):
+        self.name = name
+        self.uuid = uuid
+        self.management_cluster = list()
+
+    def add_mgmt_cluster(self, mgmt_cluster):
+        self.management_cluster.append(NSX_T_ManagementCluster(mgmt_cluster.get('name'), mgmt_cluster.get('uuid')))
+
+
+class NSX_T_ManagementCluster:
```
Comment (viennaa) [inline resources/Resourceskinds.py:105]: What is the difference between `Mgmt_Plane` and `ManagementCluster`?

**Review by viennaa — COMMENTED** [review]
Like the slim implementation, the framework pays off big time here, well done👍 .

Couple of questions are inline where I need some clarification.

File: InventoryBuilder.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ def __init__(self, atlas_config, port, sleep, timeout):
         self._user = os.environ["USER"]
         self._password = os.environ["PASSWORD"]
         self.vcenter_dict = dict()
+        self.nsx_t_dict = dict()
```
[the remark this answers is not in the document]
Comment (richardtief) [reply InventoryBuilder.py:23]: yes, will apply this to all the nsxt naming.

File: tools/Vrops.py [hunk 3]
```diff
@@ -59,19 +59,21 @@ def get_adapter(self, target: str, token: str) -> (str, str):
                                     headers=headers)
         except Exception as e:
             logger.error(f'Problem connecting to {target} - Error: {e}')
-            return name, uuid
+            return []
```
[the remark this answers is not in the document]
Comment (richardtief) [reply tools/Vrops.py:62]: Make sense. Thank you!

File: tools/Vrops.py [hunk 4]
```diff
@@ -59,19 +59,21 @@ def get_adapter(self, target: str, token: str) -> (str, str):
...
 
         if response.status_code == 200:
             for resource in response.json()["adapterInstancesInfoDto"]:
-                name = resource["resourceKey"]["name"]
-                uuid = resource["id"]
-
+                ada = dict()
+                ada['name'] = resource["resourceKey"]["name"]
+                ada['uuid'] = resource["id"]
+                adapter.append(ada)
```
[the remark this answers is not in the document]
Comment (richardtief) [reply tools/Vrops.py:69]: That's a very good idea. We could do that for all our resources to build the objects right after the responses.

File: resources/Resourceskinds.py (hunk 1 above)
Comment (richardtief) [reply resources/Resourceskinds.py]: The `NSXT_Mgmt_Plane` is like the vSphere World for the vcenter adapter. This is introduced to have a parent object for all nsxt adapters.

File: InventoryBuilder.py [hunk 5]
```diff
@@ -190,37 +196,43 @@ def query_vrops(self, vrops, vrops_short_name):
 
         logger.info(f'##########  Collecting resources {vrops_short_name}... ##########')
 
-        vcenter = self.create_resource_objects(vrops, token)
+        vcenter = self.create_vcenter_objects(vrops, token)
         if not vcenter:
             return False
```
[the remark this answers is not in the document]
Comment (richardtief) [reply InventoryBuilder.py]: Solved.

**Review by christopherhans — APPROVED** [review]
Even though it brings ~150 new lines of code, it easy to read/understand. Impressive.

**Pull request merged**